File: day 2/part1.py
inputs: list[tuple[str, str]] = [x[:-1].partition(': ')[0::2] for x in open('./day 2/input.txt').readlines()]
valid = 0
for x in inputs:
    bounds, _,  character = x[0].partition(' ')
    lowerbound, _ , upperbound = (int(x) if x != '-' else _ for x in bounds.partition('-'))
    if lowerbound <= x[1].count(character) <= upperbound:
        valid += 1

print(valid)

File: day 1/part1.js
const fs = require('fs');
const inputs = fs.readFileSync('./day 1/input.txt', 'utf8').split('\n').map(x => parseInt(x, 10));
for (var i = 0; i < inputs.length; i++) {
  for (var j = i + 1; j < inputs.length; j++) {
    if (inputs[i] + inputs[j] === 2020) {
      console.log(inputs[i] * inputs[j]);
      break;
    }
  }
}
